DancingGoat/Helpers/SampleDeliveryClient.cs

```diff
@@ -0,0 +1,160 @@
+            return _deliveryClient.GetItemJsonAsync(codename, parameters);
+        }
+
+        public Task<JObject> GetItemsJsonAsync(params string[] parameters)
+        {
+            parameters.ToList().Add(_sampleSiteFilter.FirstOrDefault().GetQueryStringParameter());
+            return _deliveryClient.GetItemsJsonAsync(parameters);
+        }
+
+        public Task<DeliveryItemResponse> GetItemAsync(string codename, params IQueryParameter[] parameters)
+        {
+            var p = parameters.Concat(_sampleSiteFilter);
+            return _deliveryClient.GetItemAsync(codename, p);
+        }
+
+        public Task<DeliveryItemResponse<T>> GetItemAsync<T>(string codename, params IQueryParameter[] parameters)
+        {
+            var p = parameters.Concat(_sampleSiteFilter);
+            return _deliveryClient.GetItemAsync<T>(codename, p);
+        }
+
+        public Task<DeliveryItemResponse> GetItemAsync(string codename, IEnumerable<IQueryParameter> parameters)
+        {
+            parameters.ToList().Add(_sampleSiteFilter.FirstOrDefault());
+            return _deliveryClient.GetItemAsync(codename, parameters);
+        }
+
+        public Task<DeliveryItemResponse<T>> GetItemAsync<T>(string codename, IEnumerable<IQueryParameter> parameters = null)
+        {
+            var p = parameters == null ? _sampleSiteFilter : parameters.Concat(_sampleSiteFilter);
+            return _deliveryClient.GetItemAsync<T>(codename, p);
+        }
+
+        public Task<DeliveryItemListingResponse> GetItemsAsync(params IQueryParameter[] parameters)
+        {
+            var p = parameters.Concat(_sampleSiteFilter);
+            return _deliveryClient.GetItemsAsync(p);
+        }
+
+        public Task<DeliveryItemListingResponse> GetItemsAsync(IEnumerable<IQueryParameter> parameters)
+        {
+            var p = parameters.Concat(_sampleSiteFilter);
+            return _deliveryClient.GetItemsAsync(p);
+        }
+
+        public Task<DeliveryItemListingResponse<T>> GetItemsAsync<T>(params IQueryParameter[] parameters)
+        {
+            var p = parameters.Concat(_sampleSiteFilter);
+            return _deliveryClient.GetItemsAsync<T>(p);
+        }
+
+        public Task<DeliveryItemListingResponse<T>> GetItemsAsync<T>(IEnumerable<IQueryParameter> parameters)
+        {
+            var p = parameters.Concat(_sampleSiteFilter);
+            return _deliveryClient.GetItemsAsync<T>(p);
+        }
+
+        public Task<JObject> GetTypeJsonAsync(string codename)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<JObject> GetTypesJsonAsync(params string[] parameters)
+        {
+            parameters.ToList().Add(_sampleSiteFilter.FirstOrDefault().GetQueryStringParameter());
+            return _deliveryClient.GetTypesJsonAsync(parameters);
+        }
+
+        public Task<ContentType> GetTypeAsync(string codename)
+        {
+            throw new NotImplementedException();

```

Comment:
Why are some of the methods not implemented? This would provide a nasty experience if someone uses it, and we don't update this ...